### DISCUSSION TITLE
Supports golang.org/x/exp/slog `Handler` interface

### DISCUSSION THREAD

**Discussion by septemhill** [uber-go/zap#1179] [Ideas]
Would we like to support (implement) `Handler` interface in [slog](https://golang.org/x/exp/slog) ?

**Comment by abhinav** [comment]
Yes, we will likely support the Handler interface in Zap if the proposal that comes out of [this discussion](https://github.com/golang/go/discussions/54763) gets approved.

**Reply by knight42** [reply]
@abhinav Hi, as to https://github.com/golang/go/issues/56345 the proposal is likely to get approved, is it desirable to implement the `slog.Handler` interface in zap now? If so, I would like to work on it.

**Reply by abhinav** [reply]
Hi, @knight42! Thanks for your interest!

Yes, it's desirable to implement the slog.Handler interface once the proposal is accepted, however there are some asterisks:
We do not want Zap's standard packages to take a dependency on golang.org/x/exp/slog. We want Zap's main APIs to depend on "log/slog" only.
So this would not be shipped in any stable packages (zap, zapcore, zapio, zaptest) until the feature is released in a new version of Go. (And it would need to be build-tagged.)
However, so that we can experiment and iterate with it earlier, we think it's okay to implement it in a exp/zapslog subpackage; it's okay if that subpackage depends on golang.org/x/exp/slog.

CC @mway @sywhang @prashantv

**Reply by knight42** [reply]
That's great! I will file a PR in a few days.
